src/main/java/at/hannibal2/skyhanni/data/GetFromSackData.kt

```diff
@@ -0,0 +1,130 @@
+package at.hannibal2.skyhanni.data
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.GetFromSackAPI
+import at.hannibal2.skyhanni.data.jsonobjects.repo.SacksJson
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.LorenzToolTipEvent
+import at.hannibal2.skyhanni.events.RepositoryReloadEvent
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NumberUtil.isInt
+import at.hannibal2.skyhanni.utils.PrimitiveItemStack
+import at.hannibal2.skyhanni.utils.PrimitiveItemStack.Companion.makePrimitiveStack
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.util.LinkedList
+import java.util.Queue
+import kotlin.time.Duration.Companion.seconds
+
+object GetFromSackData {
+
+    private val config get() = SkyHanniMod.feature.inventory
+
+    private val minimumDelay = 1.7.seconds
+
+    private val queue: Queue<PrimitiveItemStack> = LinkedList()
+    private val inventoryMap = mutableMapOf<Int, List<PrimitiveItemStack>>()
+
+    private var lastTimeOfCommand = SimpleTimeMark.farPast()
+
+    var sackList = emptyList<NEUInternalName>()
+        private set
+
+    /** Do not use this use [GetFromSackAPI.get] instead*/
+    fun addToQueue(items: List<PrimitiveItemStack>) = queue.addAll(items)
+
+    /** Do not use this use [GetFromSackAPI.slot] instead*/
+    fun addToInventory(items: List<PrimitiveItemStack>, slotId: Int) = inventoryMap.put(slotId, items)
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (queue.isNotEmpty() && lastTimeOfCommand.passedSince() >= minimumDelay) {
+            val item = queue.poll()
+            LorenzUtils.sendCommandToServer("gfs ${item.name.asString()} ${item.amount}")
+            lastTimeOfCommand = SimpleTimeMark.now()
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inventoryMap.clear()
+    }
+
+    @SubscribeEvent
+    fun onSlotClicked(event: GuiContainerEvent.SlotClickEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (event.clickedButton != 1) return // right click
```

Comment:
The comment feels confusing to me. do we filter right clicks here or do we only want right clicks below this line?